--- rc.orig	Wed Oct 18 22:44:05 2023
+++ rc	Fri Oct 20 05:25:30 2023
@@ -186,9 +186,17 @@
 	# filesystems, otherwise record which ones are mounted read-only.
 	for _dkdev in $(df /usr/{lib,libexec} $_relink |
 	    sed '1d;s/ .*//' | sort -u); do
-		_mp=$(mount -t ffs | grep "^$_dkdev") || return
+		if mount -t ffs | grep -q "^$_dkdev"; then
+			_mp=$(mount -t ffs | grep "^$_dkdev")
+		elif mount | egrep -q "(tmpfs|mfs).*/ram"; then
+			_mp=$($mount | egrep "/ram")
+		else
+			return
+		fi
+
 		if [[ $_mp == *read-only* ]]; then
-			_ro_list="$_ro_list ${_mp%% *}"
+			echo "fuguita: can't reorder libraries at this boot mode"
+			return
 		fi
 	done
 
@@ -422,8 +430,9 @@
 # Re-mount the root filesystem read/writeable. (root on nfs requires this,
 # others aren't hurt.)
 mount -uw /
-chmod og-rwx /bsd
-ln -fh /bsd /bsd.booted
+# KARL disabled
+#chmod og-rwx /bsd
+#ln -fh /bsd /bsd.booted
 
 rm -f /fastboot
 
@@ -662,7 +671,9 @@
 
 # Re-link the kernel, placing the objects in a random order.
 # Replace current with relinked kernel and inform root about it.
-/usr/libexec/reorder_kernel &
+echo 'fuguita: disabled KARL because of read-only media'
+echo
+#/usr/libexec/reorder_kernel &
 
 date
 exit 0
