--- special.orig	Thu Oct 10 00:49:46 2024
+++ special	Sat Oct 12 17:06:43 2024
@@ -12,6 +12,9 @@
 mem		type=char mode=0640 uname=root gname=kmem
 ..	#dev
 
+sbin		type=link
+ram		type=dir mode=0755 uname=root gname=wheel
+
 etc		type=dir mode=0755 uname=root gname=wheel
 acme-client.conf	type=file mode=0644 uname=root gname=wheel optional
 bgpd.conf	type=file mode=0600 uname=root gname=wheel optional
@@ -128,9 +131,6 @@
 authorized_keys	type=file mode=0600 uname=root gname=wheel
 ..	#root/.ssh
 ..	#root
-
-sbin		type=dir mode=0755 uname=root gname=wheel ignore
-..	#sbin
 
 usr		type=dir mode=0755 uname=root gname=wheel
 bin		type=dir mode=0755 uname=root gname=wheel ignore
