--- special.orig	Thu Apr 13 19:50:24 2023
+++ special	Fri Apr 14 20:07:23 2023
@@ -12,6 +12,9 @@
 mem		type=char mode=0640 uname=root gname=kmem
 ..	#dev
 
+sbin		type=link
+ram		type=dir mode=0755 uname=root gname=wheel
+
 etc		type=dir mode=0755 uname=root gname=wheel
 acme-client.conf	type=file mode=0644 uname=root gname=wheel optional
 bgpd.conf	type=file mode=0600 uname=root gname=wheel optional
@@ -130,9 +133,6 @@
 authorized_keys	type=file mode=0600 uname=root gname=wheel
 ..	#root/.ssh
 ..	#root
-
-sbin		type=dir mode=0755 uname=root gname=wheel ignore
-..	#sbin
 
 usr		type=dir mode=0755 uname=root gname=wheel
 bin		type=dir mode=0755 uname=root gname=wheel ignore
