--- RDROOT.orig	Sat Oct 12 20:32:50 2024
+++ RDROOT	Sat Oct 12 20:32:50 2024
@@ -39,7 +39,7 @@
 option		UDF		# UDF (DVD) file system
 option		MSDOSFS		# MS-DOS file system
 option		FIFO		# FIFOs; RECOMMENDED
-#option		TMPFS		# efficient memory file system
+option		TMPFS		# efficient memory file system
 option		FUSE		# FUSE
 
 option		SOCKET_SPLICE	# Socket Splicing for TCP and UDP
@@ -78,7 +78,7 @@
 pseudo-device	nmea	1	# NMEA 0183 line discipline
 pseudo-device	msts	1	# MSTS line discipline
 pseudo-device	endrun	1	# EndRun line discipline
-pseudo-device	vnd	4	# vnode disk devices
+pseudo-device	vnd	6	# vnode disk devices
 pseudo-device	ksyms	1	# kernel symbols device
 pseudo-device	kstat		# kernel statistics device
 
