--- RDROOT.orig	Thu Oct 10 16:13:31 2024
+++ RDROOT	Thu Oct 10 16:13:31 2024
@@ -10,7 +10,7 @@
 # device.
 
 machine		i386
-include		"../../../conf/GENERIC"
+include		"../../../conf/RDROOT"
 maxusers	80			# estimated number of users
 
 option		USER_PCICONF	# user-space PCI configuration
@@ -22,7 +22,12 @@
 option		SUSPEND
 option		HIBERNATE
 
-config		bsd	swap generic
+config		bsd root on rd0a swap on wd0b and sd0b
+option		RAMDISK_HOOKS
+option		MINIROOTSIZE=8192
+option		NKPTP=6
+
+pseudo-device   rd              1 # ramdisk
 
 mainbus0 at root
 
