--- GENERIC	Thu May  1 13:35:16 2025
+++ RDROOT	Fri May  2 15:25:28 2025
@@ -10,7 +10,7 @@
 # device.
 
 machine		amd64
-include		"../../../conf/GENERIC"
+include		"../../../conf/RDROOT"
 maxusers	80			# estimated number of users
 
 option		USER_PCICONF	# user-space PCI configuration
@@ -22,7 +22,12 @@
 option		SUSPEND
 option		HIBERNATE
 
-config		bsd	swap generic
+config		bsd root on rd0a swap on wd0b and sd0b
+option		RAMDISK_HOOKS
+option		MINIROOTSIZE=8192
+option		NKPTP=6
+
+pseudo-device   rd              1 # ramdisk
 
 mainbus0 at root
 
