--- RDROOT.orig	Fri Apr 12 09:59:44 2024
+++ RDROOT	Fri Apr 12 09:59:44 2024
@@ -40,7 +40,7 @@
 option		UDF		# UDF (DVD) file system
 option		MSDOSFS		# MS-DOS file system
 option		FIFO		# FIFOs; RECOMMENDED
-#option		TMPFS		# efficient memory file system
+option		TMPFS		# efficient memory file system
 option		FUSE		# FUSE
 
 option		SOCKET_SPLICE	# Socket Splicing for TCP and UDP
@@ -79,7 +79,7 @@
 pseudo-device	nmea	1	# NMEA 0183 line discipline
 pseudo-device	msts	1	# MSTS line discipline
 pseudo-device	endrun	1	# EndRun line discipline
-pseudo-device	vnd	4	# vnode disk devices
+pseudo-device	vnd	6	# vnode disk devices
 pseudo-device	ksyms	1	# kernel symbols device
 pseudo-device	kstat		# kernel statistics device
 
